File: src/demos/ps/maps/Aukba.java
package demos.ps.maps;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static demos.ps.oo.PSGame.EntFinish;
import static demos.ps.oo.PSGame.EntStart;
import static demos.ps.oo.PSGame.Shop;
import static demos.ps.oo.PSGame.getItem;
import static demos.ps.oo.PSGame.getString;
import static demos.ps.oo.PSGame.getYesNo;
import static demos.ps.oo.PSGame.mapswitch;
import demos.ps.oo.Dungeon;
import demos.ps.oo.Item;
import demos.ps.oo.PSGame;
import demos.ps.oo.PSLibItem.OriginalItem;
import demos.ps.oo.PSMenu;
import demos.ps.oo.PSMenu.DezoType;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.Scene;
import demos.ps.oo.PSMenu.SpecialEntity;

public class Aukba {

	private static final Logger log = LogManager.getLogger(Aukba.class);
	
	public static void hospital() {
		PSMenu.startScene(Scene.HOSPITAL_VILLAGE, EntityType.DEZO, DezoType.REGULAR);
		PSGame.Hospital(3); // Pricey!
		PSMenu.endScene();
	}

	public static void church() {
		PSMenu.startScene(Scene.CHURCH_VILLAGE, EntityType.DEZO, DezoType.REGULAR);
		PSGame.Church(3); // Pricey!
		PSMenu.endScene();
	}
	
	public static void hand_shop() {
		PSMenu.startScene(Scene.SHOP_HAND_VILLAGE, EntityType.DEZO, DezoType.REGULAR);
		Shop(getString("Shop_Tool_Welcome"), true, new Item[]{	getItem(OriginalItem.Inventory_Flash),
																	getItem(OriginalItem.Inventory_Escape_Cloth),
																	getItem(OriginalItem.Inventory_TranCarpet)});
		PSMenu.endScene();
	}	

	public static void icedigger_shop() {
		PSMenu.startScene(Scene.SHOP_HAND_VILLAGE, EntityType.DEZO, DezoType.REGULAR);
		Shop(getString("Shop_Tool_Welcome"), true, new Item[]{	getItem(OriginalItem.Vehicle_IceDecker)});
		PSMenu.endScene();
	}	

	public static void lefthouse1() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Left_House_1"));
		PSMenu.endScene();
	}
	public static void lefthouse2() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Left_House_2"));
		PSMenu.endScene();
	}
	public static void lefthouse3() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Left_House_3"));
		PSMenu.endScene();
	}
	public static void lefthouse4() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Left_House_4"));
		PSMenu.endScene();
	}
	public static void lefthouse5() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		if(PSMenu.Prompt(getString("Aukba_Left_House_5"), getYesNo()) == 1) {
			PSMenu.StextLast(getString("Aukba_Left_House_5Yes"));
		} else {
			PSMenu.StextLast(getString("Aukba_Left_House_5No"));
		}
		PSMenu.endScene();
	}
	public static void lefthouse6() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Left_House_6"));
		PSMenu.endScene();
	}

	public static void righthouse1() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_1"));
		PSMenu.endScene();
	}
	public static void righthouse2() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_2"));
		PSMenu.endScene();
	}
	public static void righthouse3() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_3"));
		PSMenu.endScene();
	}
	public static void righthouse4() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_4"));
		PSMenu.endScene();
	}
	public static void righthouse5() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_5"));
		PSMenu.endScene();
	}

	public static void righthouse6() {
		PSMenu.startScene(Scene.VILLAGE_HOUSE, EntityType.DEZO, DezoType.REGULAR);
		PSMenu.Stext(getString("Aukba_Right_House_6"));
		PSMenu.endScene();
	}

	public static void ent1() {
		EntStart();
		PSMenu.Stext(getString("Aukba_Left_Ent"));
		EntFinish();
	}
	public static void ent2() {
		EntStart();
		PSMenu.Stext(getString("Aukba_Right_Ent"));
		EntFinish();
	}

	public static void tunnel() {
		mapswitch(Dungeon.AUKBA_TUNNEL_OUT);
	}
	
	
}

File: src/demos/ps/maps/Aukba.ts
/**
 * Aukba Village Script
 * TypeScript port of Aukba.java
 */

import { PSGame } from '../PSGame';
import { Dungeon } from '../game/Dungeon';
import { OriginalItem } from '../game/PSLibItem';
import { PSSceneType, EntityType, EntityClothes, DezoType, PSMenu } from '../PSMenu';
import { PSMenuShop } from '../PSMenuShop';

// DezoType values act as the "clothes" frame offset, same numeric slot as EntityClothes.
const dezo = (d: DezoType): EntityClothes => d as unknown as EntityClothes;

export class Aukba {

  public static async hospital(): Promise<void> {
    await PSMenu.startScene(PSSceneType.HOSPITAL_VILLAGE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSGame.Hospital(3); // Pricey!
    await PSMenu.endScene();
  }

  public static async church(): Promise<void> {
    await PSMenu.startScene(PSSceneType.CHURCH_VILLAGE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSGame.Church(3); // Pricey!
    await PSMenu.endScene();
  }

  public static async hand_shop(): Promise<void> {
    await PSMenu.startScene(PSSceneType.SHOP_HAND_VILLAGE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenuShop.Shop(PSGame.getString("Shop_Tool_Welcome"), true, PSGame.getParty(), [
      PSGame.getItem(OriginalItem.Inventory_Flash),
      PSGame.getItem(OriginalItem.Inventory_Escape_Cloth),
      PSGame.getItem(OriginalItem.Inventory_TranCarpet)
    ]);
    await PSMenu.endScene();
  }

  public static async icedigger_shop(): Promise<void> {
    await PSMenu.startScene(PSSceneType.SHOP_HAND_VILLAGE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenuShop.Shop(PSGame.getString("Shop_Tool_Welcome"), true, PSGame.getParty(), [
      PSGame.getItem(OriginalItem.Vehicle_IceDecker)
    ]);
    await PSMenu.endScene();
  }

  public static async lefthouse1(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Left_House_1"));
    await PSMenu.endScene();
  }

  public static async lefthouse2(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Left_House_2"));
    await PSMenu.endScene();
  }

  public static async lefthouse3(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Left_House_3"));
    await PSMenu.endScene();
  }

  public static async lefthouse4(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Left_House_4"));
    await PSMenu.endScene();
  }

  public static async lefthouse5(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    if (await PSMenu.Prompt(PSGame.getString("Aukba_Left_House_5"), PSGame.getYesNo()) === 1) {
      await PSMenu.StextLast(PSGame.getString("Aukba_Left_House_5Yes"));
    } else {
      await PSMenu.StextLast(PSGame.getString("Aukba_Left_House_5No"));
    }
    await PSMenu.endScene();
  }

  public static async lefthouse6(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Left_House_6"));
    await PSMenu.endScene();
  }

  public static async righthouse1(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_1"));
    await PSMenu.endScene();
  }

  public static async righthouse2(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_2"));
    await PSMenu.endScene();
  }

  public static async righthouse3(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_3"));
    await PSMenu.endScene();
  }

  public static async righthouse4(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_4"));
    await PSMenu.endScene();
  }

  public static async righthouse5(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_5"));
    await PSMenu.endScene();
  }

  public static async righthouse6(): Promise<void> {
    await PSMenu.startScene(PSSceneType.VILLAGE_HOUSE, EntityType.DEZO, dezo(DezoType.REGULAR));
    await PSMenu.Stext(PSGame.getString("Aukba_Right_House_6"));
    await PSMenu.endScene();
  }

  public static async ent1(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Aukba_Left_Ent"));
    PSGame.EntFinish();
  }

  public static async ent2(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Aukba_Right_Ent"));
    PSGame.EntFinish();
  }

  public static async tunnel(): Promise<void> {
    await PSGame.mapswitchToDungeon(Dungeon.AUKBA_TUNNEL_OUT);
  }
}
